projects/demo-maps/src/app/route-components/route-example-custom-layers/customRenderers/threejs_renderer.ts

```diff
@@ -109,6 +109,9 @@ export class ThreeJsRenderer extends LayerRenderer<VectorLayer> {
         return this.canvas;
     }
 
+    renderDeclutter(frameState: FrameState) {
```

Comment:
> What do these functions without code?
> I also see they sometime in the OpenLayers code:
 Yeah, those are required when using version 6.5 or higher.  The concrete point is [here](https://github.com/openlayers/openlayers/commit/8e862766fcb49553b644f7f82a01e54b4bcaa68e#diff-4b51c48a2f82fb303337b99ae74dfa6ebc40a8dd5af81e6c248a25546b82ef44R130)

As long as the `Layer` (not the `LayerRenderer`, by the way!) has a `getDeclutter` field, openlayers will call it's `renderDeclutter` method. Every `BaseVectorLayer` does indeed have a `getDeclutter` method, because they do want to potentially trim the rendered output a bit when mutliple features overlap. 

In our WebGL-renderers we have simply not yet implemented such decluttering. We might do this at some point, but it's really a bit of work and probably only worth it when real need arises.

Reply:
PS: I've updated the docs to add some explanation on that `renderDeclutter` method.